pkg/cluster/internal/providers/common/util.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+// AddressPortProtocolMap is used to keep track of declared listen address,
+// port and protocol in node extra port mappings.
+type AddressPortProtocolMap map[string]map[int32]map[string]struct{}
+
+// IsAddrPortProtocolReconfigured checks if a combination of listen address, port and protocol exists in the given map.
+// Also takes care of special cases like 0.0.0.0 wildcard address.
+func IsAddrPortProtocolReconfigured(addrPortProtocolMap AddressPortProtocolMap, addr string, port int32, protocol string) bool {
```

Comment:
Instead of exporting in common and doing in the providers, we should do this at config validation time?

Reply:
That make sense, will update it.